src/platform/android/InetInterfaceImpl.cpp

```diff
@@ -0,0 +1,159 @@
+/*
+ *
+ *    Copyright (c) 2020-2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <inet/InetInterfaceImpl.h>
+
+#if CHIP_SYSTEM_CONFIG_USE_SOCKETS && CHIP_SYSTEM_CONFIG_USE_BSD_IFADDRS
+#include <ifaddrs.h>
+#include <lib/support/CHIPMem.h>
+#include <lib/support/CodeUtils.h>
+#include <net/if.h>
+namespace chip {
+namespace Inet {
+
+using namespace chip;
+
+void if_freenameindexImpl(struct if_nameindex * inArray)
```

Comment:
Can we just name this `if_freenameindex`, put it in the global namespace, and not need the "Impl" shimming?  Similar for `if_nameindex`.

That is, is Android just missing these pieces, in which case we should just implement them, or are they there but wrong?

Reply:
The function exists, but it did not work, so it was reimplemented. Therefore, implementation with the same function name is not possible. Could you recommend an suitable function name?

Reply:
If they exist but are broken, then these names seem fine.  Thank you for the clarification!